### PR TITLE
added Gaussian and Matlab Test 

### PR DESCRIPTION
**Pull request by Lewih**
Added 2 tests for Antwerpen.

Enabled only on Leibniz and Vaughan.

Added in order to share test base.

### CODE REVIEW

**Comment by Lewih** [comment]
I added Matlab support for Hydra and Genius. (the init of Matlab takes a lot of time on hydra).

Gaussian support for hydra is in draft: right permissions are required to load the gaussian module.
Missing performance reference and execution test.

**Comment by Lewih** [comment]
@smoors may u get reference for gaussianTest on hydra?

File: tests/apps/gaussian/gaussian.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    @run_after('setup')
+    def set_num_cpus(self):
+        if self.current_system.name == 'leibniz':
+            self.num_cpus_per_task = 28
+        elif self.current_system.name == 'vaughan':
+            self.num_cpus_per_task = 64
+        elif self.current_system.name == 'hydra':
+            self.num_cpus_per_task = 40
+            self.job.options = ["--partition=skylake,skylake_mpi", "--exclusive"]
+            self.modules = ['Gaussian/G16.B.01']
```
Comment (smoors) [inline tests/apps/gaussian/gaussian.py:56]: we recommend the A.03 version as it is faster: it is self-compiled, whereas B.01 is a binary.
```suggestion
            self.modules = ['Gaussian/G16.A.03-intel-2017b']
```

File: tests/apps/gaussian/gaussian.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        if self.current_system.name == 'leibniz':
+            self.num_cpus_per_task = 28
+        elif self.current_system.name == 'vaughan':
+            self.num_cpus_per_task = 64
+        elif self.current_system.name == 'hydra':
+            self.num_cpus_per_task = 40
+            self.job.options = ["--partition=skylake,skylake_mpi", "--exclusive"]
+            self.modules = ['Gaussian/G16.A.03-intel-2017b']
+
+        self.executable = f'time g16 < mich-{self.current_system.name}.com'
```
Comment (smoors) [inline tests/apps/gaussian/gaussian.py:58]: @Lewih it is not necessary to have a separate input file for each cluster.
you can specify extra options for the number of processes and memory, e.g. (for `self.memory` we should probably specify per-cluster values):
```suggestion
        self.executable = f'time g16 -p={self.num_cpus_per_task} -m={self.memory}  < mich-{self.current_system.name}.com'
```

Comment (Lewih) [reply tests/apps/gaussian/gaussian.py:58]: Very Interesting, so one file is enough. I will fix it later today.

**Comment by smoors** [comment]
> @smoors may u get reference for gaussianTest on hydra?

just tested, got 21.5 minutes in hydra-skylake.

File: tests/apps/gaussian/gaussian.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+@rfm.simple_test
+class GaussianCPUTest(GaussianBaseTest):
+    def __init__(self):
+        super().__init__()
+        self.valid_systems = ['leibniz:single-node',
+                              'vaughan:single-node',
+                              'hydra:single-node']
+        self.reference = {
+            'leibniz:single-node': {
+                'time': (33.0, -0.5, 0.5, 'minutes/percentageOfMinute'),
```
Comment (smoors) [inline tests/apps/gaussian/gaussian.py:37]: what do you mean with 'minutes/percentageOfMinute'? probably less confusing if we just use 'minutes'?

Comment (Lewih) [reply tests/apps/gaussian/gaussian.py:37]: Ah, now I remember , silly me. It is minutes.

File: tests/apps/gaussian/gaussian.py (hunk 2 above)
Comment (Lewih) [reply tests/apps/gaussian/gaussian.py:62]: I therefore assume the first two line in the `.com` files can be safely deleted
e.g
```
%cpu=0-27
%mem=109GB
```

Comment (smoors) [reply tests/apps/gaussian/gaussian.py:62]: yes

Comment (Lewih) [reply tests/apps/gaussian/gaussian.py]: Added. I used `-c` instead of `-p`, see [here](https://gaussian.com/options/) for reference.

**Comment by Lewih** [comment]
It should work.

With more performance data we will be able to fine tune the performance metrics.

**Review by smoors — APPROVED** [review]
lgtm

**Pull request merged**